internal/ziputils/ziputils.go

```
@@ -37,48 +57,37 @@ func ProcessZipFile(zipPath string) ([]string, error) {
-		// Check if there are nested directories
-		if filepath.Dir(f.Name) != "." && filepath.Dir(f.Name) != "/" {
-			hasNestedDirs = true
-			break
-		}
-
-		// Check file extension
-		if strings.ToLower(filepath.Ext(f.Name)) != ".json" {
-			hasNonJsonFiles = true
-			break
+		if !allowedZipExtensions[strings.ToLower(path.Ext(f.Name))] {
+			os.RemoveAll(tempDir)
```

golangci-lint (2.12.2)

[error] Error return value of `os.RemoveAll` is not checked

(errcheck)